src/decaf.cpp

```diff
@@ -2,3 +2,26 @@
 #include "decaf.h"
 
 #include <llvm/IR/Instructions.h>
+
+namespace decaf {
+  void Interpreter::execute() {
+    ExecutionResult exec;
+
+    do {
+      StackFrame& frame = ctx->stack_top();
+
+      DECAF_ASSERT(
+        frame.current != frame.current_block->end(),
+        "Instruction pointer ran off end of block.");
+
+      llvm::Instruction& inst = *frame.current;
```

Comment:
Nit: use arrow syntax
```suggestion
      llvm::Instruction& inst = frame->current;
```

Reply:
We're dereferencing the iterator here (i.e. `frame.current`) not `frame`.